src/sourcemap.rs

```diff
@@ -163,39 +163,49 @@ impl SourceMap {
     }
 
     /// Generate a lookup table, it will be used at `lookup_token` or `lookup_source_view_token`.
-    pub fn generate_lookup_table(&self) -> Vec<(u32, u32, u32)> {
-        let mut table = self
-            .tokens
-            .iter()
-            .enumerate()
-            .map(|(idx, token)| (token.dst_line, token.dst_col, idx as u32))
-            .collect::<Vec<_>>();
-        table.sort_unstable();
-        table
+    pub fn generate_lookup_table(&self) -> Vec<LineLookupTable> {
+        if let Some(token) = self.tokens.last() {
+            let mut table: Vec<Vec<(u32, u32, u32)>> = vec![vec![]; token.dst_line as usize + 1];
+            for (idx, token) in self.tokens.iter().enumerate() {
+                table[token.dst_line as usize].push((token.dst_line, token.dst_col, idx as u32));
+            }
```

Comment:
The fix makes sense to me!

Just double checking, is it guaranteed to have `last_token.dst_line >= token.dst_line`? and in general `(dst_line, dst_column)` is always increasing in `tokens` order?
I think I've made such assumption for visualizer, but I'm just wondering it now.

nit: maybe easier to read by renaming outer `token` as `last_token` to avoid shadowing.

Reply:
> Just double checking, is it guaranteed to have last_token.dst_line >= token.dst_line? and in general (dst_line, dst_column) is always increasing in tokens order?
I think I've made such assumption for visualizer, but I'm just wondering it now.

It is not stable order, because the top ast node maybe moving to bottom.

Here i using per-sort to avoid it. Thank you catch it.